outward_assembly/overlap_graph.py

```diff
@@ -129,25 +127,28 @@ def overlap_inds(
 
     Args:
         seqs: Sequences to analyze
-        seed_seqs: List of seed sequences to find in connected components
+        subset_seqs: Dict to use to subset the above the sequences. The keys of this dict are
+            indices of seqs, corresponding to sequences that contain a seed. The values of the
+            dict are the orientation
         n_0_error: Minimum overlap length for exact match
         n_1_error: Minimum overlap length when allowing 1 mismatch
 
     Returns:
         List of indices into seqs of sequences in connected components with any seed sequence
     """
     seqs = [Seq(s) if isinstance(s, str) else s for s in seqs]
-    seed_seqs = [Seq(s) if isinstance(s, str) else s for s in seed_seqs]
 
     g = _overlap_graph(seqs, n_0_error, n_1_error)
-    # Find sequences containing seeds as substrings (is_subseq check)
-    seed_inds = [
-        i
-        for i, seq in enumerate(seqs)
-        if any(is_subseq(seed, seq, check_rc=True) for seed in seed_seqs)
-    ]
-
     components = nx.connected_components(g)
-    seed_components = [cc for cc in components if any(i in cc for i in seed_inds)]
 
-    return sorted(set().union(*seed_components)) if seed_components else []
+    # Find sequences containing seeds as substrings (is_subseq check)
+    # Get the indices of all sequences that have a seed
+
+    # Get all sets (connected components) of indices, where at least one index corresponds to a
+    # seed-containing sequence
+    seed_components: Dict[int, bool] = {}
+    for component_set in components:
+        for idx, is_reverse_complement in subset_seqs.items():
+            if idx in component_set:
+                seed_components.update({c: is_reverse_complement for c in component_set})
+    return seed_components
```

Comment:
I worry that this logic is a little inscrutable (passing around indices of seed-containing sequences and their orientations, in order to eventually RC them if needed). Since `overlap_inds` is only called in `_subset_contigs` and nowhere else, I could also move this block back out into `_subset contigs`, and then `overlap_inds` would just return a list of connected components, and not have to accept this weird data structure as an argument. (Alternatively I could make the data structure less weird and more explicit by making it a namedtuple or dataclass or something?)